include/pybind11/gil.h

```diff
@@ -137,6 +137,7 @@ class gil_scoped_acquire {
 
 class gil_scoped_release {
 public:
+    // PRECONDITION: The GIL must be held when this constructor is called.
```

Comment:
It might be a good idea to add a debug assertion that the GIL is actually held here?

Reply:
Done, thanks! I actually globally tested that change in isolation about a week ago already and it passed. Which makes me think: if the precondition is not met the failures are critical, therefore such code never actually makes it into the code base. However, the extra `assert()` is a great and easy way to guard against accidents when developing new code.